github/open_close_pr.py

```diff
@@ -0,0 +1,184 @@
+    except yaml.YAMLError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to parse YAML from {}: {}'.format(yml_path, exc))
+
+
+class AutomatedPR(object):
+
+    def __init__(self, github_token, repo):
+        github = github3.login(token=github_token)
+        github.session.mount('https://api.github.com', CacheControlAdapter())
+        self.repo = github.repository(repo['owner'], repo['name'])
+
+    def commit_new_prci_config_file(self, new_prci_config, args):
+        """
+        Updates the .freeipa-pr-ci.yaml file with the content
+        of the --prci_config provided file.
+        """
+        prci_config_file = os.path.join(args.repo_path,
+                                        FREEIPA_PRCI_CONFIG_FILE)
+
+        content = ''
+        with open(new_prci_config, 'r') as f:
+            content = f.read()
+
+        with open(prci_config_file, 'w') as f:
+            f.write(content)
+
+        repo = Repo(args.repo_path)
+        repo.git.add(FREEIPA_PRCI_CONFIG_FILE)
+        repo.git.commit('-m', DEFAULT_COMMIT_MSG)
+        repo.git.push(MYGITHUB_REMOTE_REF, args.id)
+
+    def close_older_pr(self, pr_identifier):
+        # repo.git.push('-d', MYGITHUB_REMOTE_REF, pr_identifier)
+        # repo.git.branch('-D', identifier)
+
+        refs = {r.ref:r for r in self.repo.refs()}
+
+        for pull_request in self.repo.pull_requests():
+            if '[{}]'.format(pr_identifier) in pull_request.title:
+                ref_uri = '{}{}'.format(REF_FORMAT, pull_request.head.ref)
+                ref = refs[ref_uri]
+                ref.delete()
+                logger.debug("PR #%s closed and branch deleted",
+                             pull_request.number)
+
+    def rebase_branch(self, base_branch, identifier, git_repo_path):
+        repo = Repo(git_repo_path)
+
+        try:
+            repo.git.remote('add', UPSTREAM_REMOTE_REF, REMOTE_REPO)
+        except git.exc.GitCommandError:
+            # the remote is already configured
+            pass
+
+        repo.git.fetch(UPSTREAM_REMOTE_REF)
+        repo.git.checkout('{}/{}'.format(UPSTREAM_REMOTE_REF, base_branch))
+
+        # creates new branch using the identifier as the name
+        repo.git.checkout('-b', identifier)
+        repo.git.push(MYGITHUB_REMOTE_REF, '-u', identifier)
+
+    def open_pr(self, args):
+        # before opening a new PR, we close the old one with the same
+        # identifier. The PR list shoud have only one open PR.
+        self.close_older_pr(args.id)
+
+        self.rebase_branch(args.branch, args.id, args.repo_path)
+
+        self.commit_new_prci_config_file(args.prci_config, args)
+
+        pr_title = '[{}] Nightly PR'.format(args.id)
+
+        logger.debug("A new PR against %s/%s will be created with "
+                     "the title %s", self.repo.owner.login,
+                     self.repo.source.name, pr_title)
+
+        try:
+            pr = self.repo.create_pull(pr_title, args.branch, args.id)
```

Comment:
There has to be an option to select which repo the PR should be opened against. Usually, a new branch will be created in the user's repo, but the nightly PR will be opened against official freeipa repo. This only allows me to open a PR against my own repo, which isn't very useful, since no PR CI machinery is running there.